File: nim/grains/grains.nim
import math

proc onSquare*(square: int): uint64 =
  if square < 1 or square > 64:
    raise newException(ValueError, "Invalid square")
  else:
    2'u64 ^ uint64(square - 1)

proc total*: uint64 = 
  2'u64 ^ 64 - 1

File: julia/grains/grains.jl
function on_square(square)
    (square < 1 || square > 64) && throw(DomainError(square, "Square must be between 1 and 64"))

    UInt64(2)^(square - 1)
end

function total_after(square)
    (square < 1 || square > 64) && throw(DomainError(square, "Square must be between 1 and 64"))

    2 * on_square(square) - 1
end
